lib/spack/spack/util/web.py

```diff
@@ -385,8 +384,8 @@ def fetch_url_text(url, curl: Optional[Executable] = None, dest_dir="."):
 
     tty.debug("Fetching text at {0}".format(url))
 
-    filename = os.path.basename(url)
-    path = os.path.join(dest_dir, filename)
+    filename = PurePath(url).name
+    path = os.fsdecode(PurePath(dest_dir, filename))
```

Comment:
Why use decode here instead of the fspath? 

Cast `path` on method boundaries.